### DISCUSSION TITLE
Native Queries Relationships

### DISCUSSION THREAD

**Discussion by osdiab** [hasura/graphql-engine#9869] [General]
Hi, I see that Native Queries and Logical Models were released - nice!

I was wondering, while [I see in the docs that relationships with tables appear to be supported](https://hasura.io/docs/latest/schema/postgres/native-queries/#relationships-with-tables), it seems that that isn't surfaced in the console UI (I try to add relations and it tells me that I need to make more native queries to bind the relations to, but I want to make a relation with a table). Is this the case?

Also, regarding permissioning logical models, it seems that the options for permissions are much reduced compared to normal tables, for example I want to do an `_exists` check on a totally different table based on a column value, but that doesn't show up as an option in the Hasura Console logical models UI. Is that planned to be expanded?

Thanks! Looking forward to use native queries but seems like there's still some missing pieces.

**Comment by amitava82** [comment]
Can any anyone please answer this? Is it possible to create relationship with a table within schema? I tried editing metadata yaml file manually but no luck.

**Comment by raulelba** [comment]
I hope someone could point us in some direction here. But as far I can tell while the docs already showcase how the table relationships are defined, there doesn't seem to be a way to create them form the schema. And editing the metadata doesn't seem to do it either.

At the [top of these](https://hasura.io/docs/latest/schema/postgres/native-queries/#what-are-native-queries) docs it says:
 
> SUPPORTED FEATURES
> Currently, Hasura aggregations are not supported, but you can write the aggregation yourself as part of the query.
>
> **Relationships will be supported in a future release.**

So perhaps this is still a work in progress feature. In which case it should be more obvious in the docs.

**Comment by RS-DU34** [comment]
Today it's still not possible on the console UI, but can be done via the API. The below works.
There are some caveats, eg. we can fetch the nested relations, etc... but can't use nested objects in where clauses. (ex. I can fetch get_some_stuff.MyRelatedTable.someColumn but I can't use MyRelatedTable.someColumn in the where clause of the query)
```
curl -X POST http://localhost:3080/v1/metadata \
     -H "Content-Type: application/json" \
     -H "X-Hasura-Role: admin" \
     -d '{
           "type": "mssql_track_native_query",
           "args": {
             "returns": "BuchungGrouped",
             "root_field_name": "get_some_stuff",
             "source": "My Local Dev Db",
             "type": "query",
             "arguments": {},
             "array_relationships": [],
			 "code": "SELECT col1, col2,ExactStringOfMyLocalColumnId from complex_sub_select where whatever = joins",
             "object_relationships": [
               {
                 "name": "MyRelatedTable",
                 "using": {
                   "remote_table": "MyRelatedTable",
                   "column_mapping": {
                     "ExactStringOfMyLocalColumnId": "MyRelatedTableId"
                   },
                   "insertion_order": null
                 }
               }
             ]
           }
         }'
```

**Reply by mwahlhuetter** [reply]
The above query did not work for me, I had to change the `remote_table` property to this object:
```
"remote_table": {
  "name": "<table>",
  "schema": "<schema>"
},
```
but i have not found a way and i think its not possible right now, to use this relation for permissions because you need a logical model for that.

**Comment by s-horbach** [comment]
> can't use nested objects in where clauses

I've faced exactly the same issue.

I have a table and a native query, then i manually created a relations between those according to [Tracking relationships between tables and Native Queries](https://hasura.io/docs/latest/schema/postgres/table-relationships/create/#tracking-relationships-between-tables-and-native-queries) article.

But I can query my table including native query as a nested property only if there is no `where` clause. And once I add `where` - it fails with the next error:
`"code": "unexpected",
        "internal": {
          "error": "translateBoolExp RelTargetNativeQuery"
        }`

On the other hand, if I query native query directly with  `where` clause - it works perfect 🤷‍♂️ 

P.S. Query example:
```
query MyQuery {
  MyTable(where: {MyNativeQuery: {Name: {_eq: "abc"}}}) {
    Id
    MyNativeQuery {
      Name
    }
  }
}

```

**Comment by th3coop** [comment]
I hit this as well and the where clause not working makes this feature unusable unless I'm missing a sensible workaround?  Even if there is, I'm reluctant to use NQs at this point.
